lib/hexpm/repository/packages.ex

```diff
@@ -71,4 +71,16 @@ defmodule Hexpm.Repository.Packages do
     |> Repo.all
     |> Enum.into(%{})
   end
+
+  def dependent_package_ids(name) do
+    from(rq in Requirement,
+      join: p in Package,
+      join: rl in Release,
+      select: rl.package_id,
+      distinct: rl.package_id,
+      where: rq.dependency_id == p.id,
+      where: rl.id == rq.release_id,
+      where: p.name == ^name,
+    ) |> Repo.all
+  end
```

Comment:
I think you can build this as a query, without `Repo.all`, directly inside `package.ex`. You can also rework the query so that it's composable on query based on `Package`, like these are: https://github.com/hexpm/hexpm/blob/master/lib/hexpm/repository/package.ex#L120-L121, that way you don 't need to do two queries.